### DISCUSSION TITLE
How to get parent mote of a mote?

### DISCUSSION THREAD

**Discussion by ceylanim** [contiki-ng/contiki-ng#2348] [Q&A] [answered]
Hi. 
I need to get parent mote info of a node. For example ip address. 
I am not sure but maybe I can access with current instance's dag ...
curr_instance.dag.preferred_parent...
But I couldnt. 
Thank you for help.

**Comment by arurke** [comment, marked as answer]
How about something like this (assuming you don't have multiple instances or DAGs that you want to distinguish):
```
#include "net/routing/rpl-classic/rpl.h"  // rpl-lite/rpl.h if RPL lite

uip_ipaddr_t* parent_ipaddr = NULL;
rpl_dag_t* rpl_dag = rpl_get_any_dag();
if(rpl_dag != NULL) {
  parent_ipaddr = rpl_parent_get_ipaddr(rpl_dag->preferred_parent);
}
```

**Reply by ceylanim** [reply]
it works @arurke . Thank you very much.
